homeassistant/components/vesync/services.py

```diff
@@ -0,0 +1,36 @@
+"""Support for VeSync Services."""
+
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant, ServiceCall, callback
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import DOMAIN, SERVICE_UPDATE_DEVS, VS_MANAGER
+
+
+@callback
+def async_setup_services(hass: HomeAssistant) -> None:
+    """Handle for services."""
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_UPDATE_DEVS, async_new_device_discovery
+    )
+
+
+async def async_new_device_discovery(call: ServiceCall) -> None:
+    """Discover and add new devices."""
+
+    entries = call.hass.config_entries.async_entries(DOMAIN)
+    entry = entries[0] if entries else None
+
+    if not entry:
+        raise ServiceValidationError("Entry not found")
+    if entry.state is not ConfigEntryState.LOADED:
+        raise ServiceValidationError("Entry not loaded")
+    manager = call.hass.data[DOMAIN][VS_MANAGER]
+    known_devices = list(manager.devices)
+    await manager.get_devices()
+    new_devices = [device for device in manager.devices if device not in known_devices]
+
+    if new_devices:
+        async_dispatcher_send(call.hass, "vesync_new_devices", new_devices)
```

Comment:
@cdnninja did you get a chance to look at the dispatcher mechanism?

Reply:
Not yet, hoping to spend some time on this over the weekend.